### ISSUE TITLE
Heartbeat - Scale - Y-axis

### ISSUE THREAD

**Issue by BK01** [bcgov/DBC-APIM#42]
**Skills / SW**:
- JavaScript, HTML

**Background**:
https://github.com/bcgov/dbcrss/blob/master/heartbeat/README.md

**Description**:
> Investigate updating the JavaScript to show the charts with a single Y-axis while still maintaining the general shape of the lines. 
> Fork and adjust as necessary. This repo uses both master and gh-pages branch.

- Current JavaScript files for reference:
https://github.com/bcgov/dbcrss/tree/master/heartbeat/src/d3

- Same JavaScript files, but used with heartbeat pages
https://github.com/bcgov/dbcrss/tree/gh-pages/heartbeat

**Potential references**:
https://d3js.org/
https://github.com/d3/d3/wiki/Gallery

**Comment by BK01** [comment]
No solution identified at this time that would improve the existing chart style.

**Comment by banders** [comment]
Rather than prescribing that we need to merge the data sets onto a chart with a single Y axis, perhaps it is better to consider the underlying requirement: We want a graphical display that will allow a system admin to assess the following "at a glance"

1. Is the app operating normally **or** is there an outage **or** is response time poor?
2. If response time is poor, is it due to network latency, or is it internal to the app itself?

To address question #1, let's change the blue line ("execution time") to "response time" (i.e. _total_ response time).

To address question #2 let's replace the red line ("network latency") with "execution time as fraction of response time" (i.e execution time divided by response time).  The right-hand Y axis will be changed to show the fraction (0-1).  Downward spikes will indicate above average network latency, and upward spikes will indicate above average execution time within the app itself.

We also have a third line (the purple line).  I think we should either remove it, or change the line style to make it clear which axis the line should be interpreted relative to (e.g. a dashed blue line or a dashed red line).

Thoughts?

**Comment by BK01** [comment]
@BK01 to research email alert services for Heartbeat. If value = 0, then email alert to DataBC contact. Pass email account to script as parameter from Jnks.

update: already in place for DataBC middleware using Jenkins/Nagios.

**Comment by BK01** [comment]
@banders to proceed with proposal for one Heartbeat graph (any is fine). Following review by team, may also be implemented to other graphs (geocoder and route planner)

**Comment by banders** [comment]
@BK01 @ll911 I have updated the heartbeat graphs as outlined above (mostly).

- [public geocoder](http://bcgov.github.io/dbcrss/heartbeat/geocoder_pub_latency/)
- [gated geocoder](http://bcgov.github.io/dbcrss/heartbeat/geocoder_sec_latency/)
- [router](http://bcgov.github.io/dbcrss/heartbeat/router_latency/)

All graphs have been updated because they share common code.

I have also added some new information above the graphs to clarify exactly what each metric is.

I decided rather than have different left and right axes on the same graph that I would instead put the two data sets on separate graphs (top and bottom).  I think it's easier to read this way.  I also think it is now fairly simple to quickly assess the two main questions that I outlined in my previous comment.  Let me know what you think.
